src/containers/find-offer/offer-container/OfferContainer.tsx

```diff
@@ -32,34 +35,39 @@ const OfferContainer: FC<OfferContainerProps> = ({
   const renderSquareCard =
     isMobile || (isDesktop && viewMode === CardsViewEnum.Grid)
 
-  const buttonActions = [
-    {
-      label: t('common.labels.sendMessage'),
-      handleClick: () => null
-    },
-    {
-      label: t('common.labels.viewDetails'),
-      handleClick: () => null
-    }
-  ]
-
-  const offerItems = offerCards.map((el) => (
-    <Grid item key={el._id} sm={isDesktop && isFiltersOpen ? 6 : 4}>
-      {renderSquareCard ? (
-        <AppCard>
-          <OfferCardSquare offer={el} onBookmarkClick={onBookmarkClick} />
-        </AppCard>
-      ) : (
-        <AppCard sx={{ padding: { sm: '20px', md: '30px 20px' } }}>
-          <OfferCard
-            buttonActions={buttonActions}
-            offer={el}
-            onBookmarkClick={onBookmarkClick}
-          />
-        </AppCard>
-      )}
-    </Grid>
-  ))
+  const offerItems = offerCards.map((el) => {
+    const buttonActions = [
+      {
+        label: t('common.labels.sendMessage'),
+        handleClick: () => null
+      },
+      {
+        label: t('common.labels.viewDetails'),
+        handleClick: () => null,
+        buttonProps: {
+          component: Link,
+          to: `${authRoutes.offerDetails.path}/${el._id}`
+        }
```

Comment:
Why do you use buttonProps if you have handleClick for button handle?
You can pass the link to handleClick